Add MyPy

There are four failing checks that need to be resolved before MyPy can be enabled again:

```
cuenca/resources/base.py:18: error: Too many arguments for "Resource"
cuenca/resources/transfers.py:72: error: Incompatible return value type (got "Resource", expected "Transfer")
cuenca/resources/api_keys.py:30: error: Incompatible return value type (got "Resource", expected "ApiKey")
cuenca/resources/api_keys.py:43: error: Incompatible return value type (got "Resource", expected "ApiKey")
Found 4 errors in 3 files (checked 12 source files)
```

--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ lint:
 	flake8 $(PROJECT) tests setup.py
 	$(isort) --check-only
 	$(black) --check
-#	mypy $(PROJECT) tests
+	mypy $(PROJECT) tests
 
 clean:
 	rm -rf `find . -name __pycache__`

--- a/cuenca/resources/api_keys.py
+++ b/cuenca/resources/api_keys.py
@@ -1,5 +1,5 @@
 import datetime as dt
-from typing import ClassVar, Optional
+from typing import ClassVar, Optional, cast
 
 from pydantic.dataclasses import dataclass
 
@@ -27,8 +27,7 @@ class ApiKey(Creatable, Queryable, Retrievable):
 
     @classmethod
     def create(cls) -> 'ApiKey':
-        resource = cls._create()
-        return cls(**resource.to_dict())
+        return cast('ApiKey', cls._create())
 
     @classmethod
     def deactivate(cls, api_key_id: str, minutes: int = 0) -> 'ApiKey':
@@ -41,5 +40,4 @@ class ApiKey(Creatable, Queryable, Retrievable):
         """
         url = cls._endpoint + f'/{api_key_id}'
         resp = session.delete(url, dict(minutes=minutes))
-        resource = cls._from_dict(resp)
-        return cls(**resource.to_dict())
+        return cast('ApiKey', cls._from_dict(resp))

--- a/cuenca/resources/api_keys.py
+++ b/cuenca/resources/api_keys.py
@@ -27,7 +27,8 @@ class ApiKey(Creatable, Queryable, Retrievable):
 
     @classmethod
     def create(cls) -> 'ApiKey':
-        return cls._create()
+        resource = cls._create()
+        return cls(**resource.to_dict())
 
     @classmethod
     def deactivate(cls, api_key_id: str, minutes: int = 0) -> 'ApiKey':
@@ -40,4 +41,5 @@ class ApiKey(Creatable, Queryable, Retrievable):
         """
         url = cls._endpoint + f'/{api_key_id}'
         resp = session.delete(url, dict(minutes=minutes))
-        return cls._from_dict(resp)
+        resource = cls._from_dict(resp)
+        return cls(**resource.to_dict())

--- a/cuenca/resources/base.py
+++ b/cuenca/resources/base.py
@@ -12,7 +12,7 @@ from ..validators import QueryParams
 class Resource:
     _endpoint: ClassVar[str]
 
-    def __init__(self, **kwargs):
+    def __init__(self, **kwargs):  # pragma no cover
         for attr, value in kwargs.items():
             setattr(self, attr, value)
 

--- a/cuenca/resources/base.py
+++ b/cuenca/resources/base.py
@@ -12,6 +12,10 @@ from ..validators import QueryParams
 class Resource:
     _endpoint: ClassVar[str]
 
+    def __init__(self, **kwargs):
+        for attr, value in kwargs.items():
+            setattr(self, attr, value)
+
     @classmethod
     def _from_dict(cls, obj_dict: Dict[str, Union[str, int]]) -> 'Resource':
         cls._filter_excess_fields(obj_dict)

--- a/cuenca/resources/transfers.py
+++ b/cuenca/resources/transfers.py
@@ -1,5 +1,5 @@
 import datetime as dt
-from typing import ClassVar, Optional
+from typing import ClassVar, Optional, cast
 
 from clabe import Clabe
 from pydantic import BaseModel, StrictStr
@@ -70,8 +70,7 @@ class Transfer(Creatable, Queryable, Retrievable):
             recipient_name=recipient_name,
             idempotency_key=idempotency_key,
         )
-        resource = cls._create(**req.dict())
-        return cls(**resource.to_dict())
+        return cast('Transfer', cls._create(**req.dict()))
 
     @staticmethod
     def _gen_idempotency_key(account_number: str, amount: int) -> str:

--- a/cuenca/resources/transfers.py
+++ b/cuenca/resources/transfers.py
@@ -70,7 +70,8 @@ class Transfer(Creatable, Queryable, Retrievable):
             recipient_name=recipient_name,
             idempotency_key=idempotency_key,
         )
-        return cls._create(**req.dict())
+        resource = cls._create(**req.dict())
+        return cls(**resource.to_dict())
 
     @staticmethod
     def _gen_idempotency_key(account_number: str, amount: int) -> str:

--- a/cuenca/version.py
+++ b/cuenca/version.py
@@ -1,3 +1,3 @@
-__version__ = '0.1.3'
+__version__ = '0.1.4'
 CLIENT_VERSION = __version__
 API_VERSION = '2020-03-19'

--- a/setup.cfg
+++ b/setup.cfg
@@ -15,7 +15,7 @@ force_grid_wrap=0
 combine_as_imports=True
 
 [mypy-pytest]
-ignore_missing_imports = true
+ignore_missing_imports = True
 
 [mypy-clabe]
 ignore_missing_imports = True

--- a/setup.cfg
+++ b/setup.cfg
@@ -16,6 +16,3 @@ combine_as_imports=True
 
 [mypy-pytest]
 ignore_missing_imports = True
-
-[mypy-clabe]
-ignore_missing_imports = True

--- a/setup.cfg
+++ b/setup.cfg
@@ -16,3 +16,6 @@ combine_as_imports=True
 
 [mypy-pytest]
 ignore_missing_imports = true
+
+[mypy-clabe]
+ignore_missing_imports = True

